Declare properties in classes to avoid having problems with dynamic properties

ordago opened the pull request:
Dynamic properties are going to be deprecated in 8.2

Relevant RFC: https://wiki.php.net/rfc/deprecate_dynamic_properties

The only changes are the properties added. There is no change in functionality.  It just declares properties I found that were dynamically created.

1️⃣ Is this something that is wanted/needed? Did you create a discussion about it first?

Discussion here: https://github.com/livewire/livewire/discussions/5168

2️⃣ Did you create a branch for your fix/feature? (Master branch PR's will be closed)

yes

3️⃣ Does it contain multiple, unrelated changes? Please separate the PRs out.

No, just the properties.

ordago committed:
declare properties in classes to avoid having problems with dynamic properties in php8.2

joshhanley commented:
@ordago thanks for the PR! 

Seems to be some issues with the changes. 

Can you review the failed tests and see if you can fix the issue?

Thanks!

ordago committed:
removes conflicting properties

ordago commented:
Hi @joshhanley 
The master branch gave me errors when I run it on my machine. From the type of errors I saw they seemed like the kind of errors that make the tests fail on windows but not on mac/linux. For that I missed the new ones. Sorry for that.

On the last commit I removed 3 of the properties that I had added that were making trouble.

joshhanley commented:
@ordago thanks! All looks good now 🙂

joshhanley commented:
@ordago thanks for the PR! In V3 we are going to avoid dynamic properties. As such though we're concerned about this implications of this on peoples apps.

Instead @calebporzio would prefer just to add the `#[AllowDynamicProperties]` attribute to any required classes as per https://php.watch/versions/8.2/dynamic-properties-deprecated#AllowDynamicProperties

Any chance you could update this PR to do this?

Thanks!

ordago commented:
Hi there @joshhanley. Sorry, I currently have my hands full.

joshhanley commented:
@ordago no worries! Thanks for getting this all started though 🙂
